modules/caddyhttp/rewrite/caddyfile.go

```diff
@@ -25,20 +27,116 @@ func init() {
 
 // parseCaddyfile sets up the handler from Caddyfile tokens. Syntax:
 //
-//     rewrite [<matcher>] <to>
+//     rewrite [<matcher>] [<to>] {
+//         to                <string>
+//         method            <string>
+//         strip_path_prefix <string>
+//         strip_path_suffix <string>
+//         rehandle          <bool>
+//         http_redirect     <status_code>
+//     }
 //
-// The <to> parameter becomes the new URI.
 func parseCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, error) {
-	var rewr Rewrite
+	var (
+		rewr Rewrite
+
+		toIsSet        bool
+		nextBlockIsSet bool
+		rehandleIsSet  bool
+		redirectIsSet  bool
+	)
+
 	for h.Next() {
-		if !h.NextArg() {
+		args := h.RemainingArgs()
+		switch len(args) {
+		case 0:
+		case 1:
+			toIsSet = true
+			rewr.URI = h.Val()
+		default:
 			return nil, h.ArgErr()
 		}
-		rewr.URI = h.Val()
-		if h.NextArg() {
-			return nil, h.ArgErr()
+
+		for h.NextBlock(0) {
+			nextBlockIsSet = true
+			switch h.Val() {
+			case "to":
+				if toIsSet {
+					return nil, h.Err("to is already set")
+				}
+				if !h.Args(&rewr.URI) {
+					return nil, h.ArgErr()
+				}
+			case "method":
+				if !h.Args(&rewr.Method) {
+					return nil, h.ArgErr()
+				}
+				switch rewr.Method {
+				case "GET", "HEAD", "POST", "PUT", "DELETE",
+					"CONNECT", "OPTIONS", "TRACE", "PATCH":
+				default:
+					return nil, h.Errf("unknown method '%s'", rewr.Method)
```

Comment:
It might not be a bad idea to allow arbitrary methods in case some upstream app requires them, for example, WebDAV...